sys/kern/exec.c

```diff
@@ -339,6 +356,10 @@ static int _do_execve(exec_args_t *args) {
     use_interpreter = true;
   }
 
+  uid_t uid;
+  gid_t gid;
+  bool setid = check_setid(vn, &uid, &gid);
```

Comment:
I think this implementation is prone to TOCTOU related vulnerabilities. How do we ensure nobody changed the binary in the meantime? Suppose we have a binary with `suid` or `sgid` that got replaced between lines 361 and 373 - is that possible? If so, then I'd be a good idea to leave a comment here to future readers to make them aware of it.

Reply:
Yes, it is possible. I left a comment.

Reply:
Why don't we lock the vnode then?

Reply:
It will be resolved by adding `ETXTBSY` error here and in filesystem syscalls.